test/integration/appfactory/appfactory_test.go

```diff
@@ -15,23 +15,135 @@
-		bootstrap.DefaultVerify(assert)
+	appFactory.DefineVerify(func(assert *assert.Assertions) {
+		appFactory.DefaultVerify(assert)
+
+		// check admin project
+		adminProjectID := appFactory.GetStringOutput("app_admin_project_id")
+		adminProjectApis := []string{
+			"iam.googleapis.com",
+			"cloudresourcemanager.googleapis.com",
+			"cloudbuild.googleapis.com",
+			"secretmanager.googleapis.com",
+			"serviceusage.googleapis.com",
+			"cloudbilling.googleapis.com",
+			"cloudfunctions.googleapis.com",
+			"apikeys.googleapis.com",
+			"sourcerepo.googleapis.com",
+		}
+
+		prj := gcloud.Runf(t, "projects describe %s", adminProjectID)
+		assert.Equal("ACTIVE", prj.Get("lifecycleState").String(), fmt.Sprintf("project %s should be ACTIVE", adminProjectID))
+
+		enabledAPIS := gcloud.Runf(t, "services list --project %s", adminProjectID).Array()
+		listApis := testutils.GetResultFieldStrSlice(enabledAPIS, "config.name")
+		assert.Subset(listApis, adminProjectApis, "APIs should have been enabled")
+
+		// check app infra repo
+		repositoryName := appFactory.GetStringOutput("app_infra_repository_name")
+		repoURL := fmt.Sprintf("https://source.developers.google.com/p/%s/r/%s", adminProjectID, repositoryName)
+		repoOP := gcloud.Runf(t, "source repos describe %s --project %s", repositoryName, adminProjectID)
+		assert.Equal(repoURL, repoOP.Get("url").String(), "source repo %s should have url %s", repositoryName, repoURL)
+
+		// check workspace SA access to repo
+		repoSa := fmt.Sprintf("serviceAccount:tf-cb-%s@%s.iam.gserviceaccount.com", repositoryName, adminProjectID)
+		repoIamOpts := gcloud.WithCommonArgs([]string{"--flatten", "bindings", "--filter", "bindings.role:roles/viewer", "--format", "json"})
+		repoIamPolicyOp := gcloud.Run(t, fmt.Sprintf("source repos get-iam-policy %s --project %s", repositoryName, adminProjectID), repoIamOpts).Array()[0]
+		listMembers := utils.GetResultStrSlice(repoIamPolicyOp.Get("bindings.members").Array())
+		assert.Contains(listMembers, repoSa, fmt.Sprintf("Service Account %s should have role roles/viewer on repo %s", repoSa, repositoryName))
+
+		// check cloudbuild workspace
+		// buckets
+		gcloudArgsBucket := gcloud.WithCommonArgs([]string{"--project", adminProjectID, "--json"})
+		for _, bucket := range []struct {
+			output string
+			suffix string
+		}{
+			{
+				output: "app_cloudbuild_workspace_state_bucket_name",
+				suffix: "state",
+			},
+			{
+				output: "app_cloudbuild_workspace_logs_bucket_name",
+				suffix: "logs",
+			},
+			{
+				output: "app_cloudbuild_workspace_artifacts_bucket_name",
+				suffix: "build",
+			},
+		} {
+			bucketSelfLink := appFactory.GetStringOutput(bucket.output)
+			opBucket := gcloud.Run(t, fmt.Sprintf("storage ls --buckets gs://bkt-%s-app1-%s", adminProjectID, bucket.suffix), gcloudArgsBucket).Array()
+			assert.Equal(bucketSelfLink, opBucket[0].Get("metadata.selfLink").String(), fmt.Sprintf("The bucket SelfLink should be %s.", bucketSelfLink))
```

Comment:
Same question here regarding the single element for `opBucket[0]`

Reply:
`gs://bkt-%s-app1-%s"` é a full mach on the bucket name. It is a match or an error.